### ISSUE TITLE
Remove `topology` configmap on kuadrant operator uninstall

### ISSUE THREAD

**Issue by eguzki** [Kuadrant/kuadrant-operator#896]
Link the lifecycle of the kuadrant operator with the `topology` configmap. Currently, when the operator starts, it will create and maintain the `topology` configmap in the namespace where the operator is deployed. It is missing the implementation of the removal of the `topology` configmap when the operator is deleted. 

> Note: The ConsolePlugin resource, together with the associated deployment and service, follows the life events of the `topology` configmap. When the topology is created, so it is the ConsolePlugin (on openshift). Whereas when the topology is deleted, so it is the ConsolePlugin and associated resources.  The ConsolePlugin was introduced in https://github.com/Kuadrant/kuadrant-operator/pull/884

**Comment by maleck13** [comment]
Is this essential for v1?

**Comment by eguzki** [comment]
I would say it is not.

**Comment by eguzki** [comment]
`Release` attribute now is unset

**Comment by david-martin** [comment]
I'm wondering when to actually delete the topology configmap.
Specifically, what does it mean 'when the operator is deleted'.
I'm not sure how we'd hook into that part of the lifecycle.
Perhaps we tie it to the created/deletion of the Kuadrant CR?

Thoughts @eguzki

**Comment by Boomatang** [comment]
I don't think you can (or should) tie this to the creation and deletion of the Kuadrant CR. What happens in the console plugin if that topology is not there? If you create a policy when the kuadrant CR is not present, the operator will accepted it, but raise a `MissingResource` error stating that kuadrant is not installed. The resource is in the topology at this stage. 

In a prefect world the console plugin would show any policies that have errors in the graph. I don't think that happens today. 

I think the removal has to be linked into the lifecycle of the operator. Could it be a simple as adding an ownership to the configmap that links back to the kuadrant deployment CR. 

There is a future world that is some way away, but might be worth the consideration. In the future there will be an [Open Architecture](https://github.com/Kuadrant/architecture/pull/115) that could used to remove the need for the configmap all together.

**Comment by guicassolato** [comment]
> I don't think you can (or should) tie this to the creation and deletion of the Kuadrant CR.
+1

One can have multiple resources, including Kuadrant policies, without necessarily having a Kuadrant CR. Even if Kuadrant is not fully functional without it⸺and _especially_ because it's not functional⸺this state should be reflected in the topology.

It's also possible to have multiple Kuadrant CRs in a cluster. Users will also want to see that in the topology graph.

IOW, a topology without a Kuadrant CR (or with multiple ones) communicate something to the user and therefore these two objects, Kuadrant CR and topology, should not be coupled together.

**Comment by eguzki** [comment]
I was thinking on adding an empty `topology` configmap as part of the bundle of the operator. Either Helm or OLM would delete it when the operator is being decommissioned. Initially, the configmap would be empty, having the operator fill with some content as soon as it starts.

**Comment by guicassolato** [comment]
> I was thinking on adding an empty `topology` configmap as part of the bundle of the operator. Either Helm or OLM would delete it when the operator is being decommissioned. Initially, the configmap would be empty, having the operator fill with some content as soon as it starts.

Brilliant!

**Comment by david-martin** [comment]
Alternative implementation idea from https://github.com/Kuadrant/kuadrant-operator/pull/1412#issuecomment-3078945669

Instead of creating the topology ConfigMap in the operator, we could define it as a manifest managed by Helm or OLM. This ensures proper cleanup on uninstall, assuming the user uses the same tool to remove it.

Things to consider:
* Helm requires a label for tracking – we’d need to verify it gets applied to existing ConfigMaps. If not, we can document it in the release notes.
* Test uninstall behaviour on upgraded clusters (both Helm and OLM).
* Existing operator logic that creates the ConfigMap if missing can be left for dev workflows. If we want to block that, local make targets would need updates.

Relevant PR: https://github.com/Kuadrant/kuadrant-operator/pull/1407

Related diffs:

* [diff 1](https://github.com/Kuadrant/kuadrant-operator/pull/1407/files#diff-493dbd4018c76caaa86d26da6cfe7c3b19ef5c6e601d1f64890c4d588449e0b7)
* [diff 2](https://github.com/Kuadrant/kuadrant-operator/pull/1407/files#diff-8e81cd5e3f0685991f3512a73713bb4128900ff04b3e3ebc5c54746501e9d092)

**Comment by nXtCyberNet** [comment]
hi  since the approach was already been decided by @david-martin , can i do the implimentation ?
